File: src/main/java/com/example/serveroptvisualizer/concert/ConcertController.java
package com.example.serveroptvisualizer.concert;

import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;

@Controller
@RequestMapping("/concerts")
public class ConcertController {

  private final ConcertService concertService;

  public ConcertController(ConcertService concertService) {
    this.concertService = concertService;
  }

  @GetMapping
  public String getConcerts(Model model) {
    model.addAttribute("concerts", concertService.getConcerts());
    return "concerts/list";
  }
}

File: src/main/kotlin/com/example/serveroptvisualizer/concert/ConcertController.kt
package com.example.serveroptvisualizer.concert

import org.springframework.stereotype.Controller
import org.springframework.ui.Model
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping

@Controller
@RequestMapping("/concerts")
class ConcertController(private val concertService: ConcertService) {

  @GetMapping
  fun getConcerts(model: Model): String {
    model.addAttribute("concerts", concertService.getConcerts())
    return "concerts/list"
  }
}
